packages/big-design/src/components/Checkbox/Checkbox.tsx

```diff
@@ -58,6 +66,14 @@ const RawCheckbox: React.FC<CheckboxProps & PrivateProps> = ({
     warning('label must be either a string or a CheckboxLabel component.');
   }, [disabled, hiddenLabel, id, label, labelId]);
 
+  const descriptionlink = typeof description === 'object' ? description.link : undefined;
+  const descriptionText = typeof description === 'object' ? description.text : description;
+
+  const renderedDescription = useMemo(
+    () => descriptionText && <FormControlDescription link={descriptionlink}>{descriptionText}</FormControlDescription>,
+    [descriptionlink, descriptionText],
+  );
+
```

Comment:
@capsula4 something like this:
```suggestion
  const renderedDescription = useMemo(() => {
      if (!description) {
        return null;
      }
      
      const text = typeof description === 'string' : description ? description.text;

      return <FormControlDescription link={description.link}>{text}</FormControlDescription>
    },
    [description, description.link, description.text],
  );

```

Reply:
I don't think that would work, as in order to access description.link and descripion.text would have to first check the type of description.

Reply:
🤔 I think you're right. This is what I tried and worked locally, which is basically your first commit anyways. :dunce-hat:
```tsx
  const renderedDescription = useMemo(() => {
    if (!description) {
      return null;
    }

    const text = typeof description === 'object' ? description.text : description;
    const link = typeof description === 'object' ? description.link : undefined;

    return <FormControlDescription link={link}>{text}</FormControlDescription>;
  }, [description]);
```

Reply:
saying that, I think that what I have now might be better as I'm passing the reference of link instead of description object ? 🤷‍♂ 
I also think it might be quite unlikely that those values could change but 🤷‍♂

Reply:
Even if those values change, the parent could cause a rerender on it's children.

So I did some playing around inside the performance tab, and this is what I found to actually avoid unnecessary rerenders:
```tsx
  const renderedDescription = useMemo(() => {
    if (!description) {
      return null;
    }

    if (typeof description === 'string') {
      return <FormControlDescription>{description}</FormControlDescription>;
    }

    return <FormControlDescription link={description.link}>{description.text}</FormControlDescription>;
  }, [description]);
```

Reply:
Yeah, I tried playing around more in the performance tab, but yeah, no matter what `link` is always going to cause a rerender since it's a reference, but specifically when `typeof description === 'object'`. There's no way to get around it, whether the assign the variables inside or outside the `useMemo`.

IMO, I'd rather have `text` and `link` assignments inside the `useMemo` scope, but I'll leave that up to you 🍹.

Reply:
Thanks for the taking the time in exploring the performance of both approaches! 🙇

Reply:
Just pushed the change 👍